### ISSUE TITLE
Lab 06: This request is not authorized to perform this operation using this permission.

### ISSUE THREAD

**Issue by AndreeaDan27** [MicrosoftLearning/dp-203-azure-data-engineer#80]
# Module: Transform data using Spark in Synapse Analytics
### Task: Use a Spark notebook to transform data
 [Link to source](https://microsoftlearning.github.io/dp-203-azure-data-engineer/Instructions/Labs/06-Transform-Data-with-Spark.html)

Description of issue

Repro steps:

1. I copy pasted the content of the notebook in a jupyter file

<img width="582" alt="Screenshot 2023-12-29 133404" src="https://github.com/MicrosoftLearning/dp-203-azure-data-engineer/assets/120377092/06f22b55-33eb-4bf5-991e-e57ad3bd19b5">

3. I imported it into the Synapse workspace

4. Tried to run the first cell to read the data but error came up that the request is not authorized. At this point I didn't do anything else not mentioned in the guide. 

<img width="1178" alt="Screenshot 2023-12-29 133144" src="https://github.com/MicrosoftLearning/dp-203-azure-data-engineer/assets/120377092/89a2372f-ebba-474f-a9ef-d476f7263090">

<img width="872" alt="Screenshot 2023-12-29 133211" src="https://github.com/MicrosoftLearning/dp-203-azure-data-engineer/assets/120377092/9dae7376-9e70-40b3-90d9-bf265f2aa412">

6. Tried to add the Managed Instance to the Blob Storage as Blob Contributor but didn't help

<img width="844" alt="Screenshot 2023-12-29 133322" src="https://github.com/MicrosoftLearning/dp-203-azure-data-engineer/assets/120377092/f780adc9-a4af-47a3-8342-d5440f0aaca9">

What am I doing wrong?

**Comment by Mihai-Ac** [comment]
Hi @AndreeaDan27 

I haven´t been able to reproduce this, it seems that there is something wrong with the user you are using.
Here´s what alese you could try : 
 - Grant yourself the Synapse Admin or Synapse Contributor role under Synapse Studio > Access Control tab
 - You can also try adding a role assignment in Azure Portal -> Storage Accounts -> Your Storage Account -> Access Control (IAM) -> Add role assignment

**Comment by AndreeaDan27** [comment]
Hello @Mihai-Ac!

I did both of the above recommended ones. My user was already a Synapse admin 
<img width="1033" alt="Screenshot 2024-01-03 122741" src="https://github.com/MicrosoftLearning/dp-203-azure-data-engineer/assets/120377092/dead8616-d5c9-465a-a67d-9b959a7ac1bb">

It didn't work for this lab. 

However, I moved on from this lab to lab 07 and that one worked without issues. So, I will close this issue since maybe there's something on my side specifical for that lab. Thanks.
